test/test_fullchain.py

```diff
@@ -136,20 +128,27 @@
 @pytest.mark.order(1)
 def test_treeToHist(script_runner):
 
-    if os.path.isfile("test_tree.root"):
-        # Check if ttree file is old, and remove to avoid getting caught offguard by changes
-        if time.time() - os.path.getctime("test_tree.root") > 86400.: # 24 hours in seconds
-            os.remove("test_tree.root")
+    #make directory
+    if not os.path.isdir("data/hf_test"):
+        os.mkdir("data/hf_test")
+    
 
+    if os.path.isfile("data/hf_test/test_tree.root"):
+        # Check if ttree file is old, and remove and remake to avoid getting caught offguard by changes
+        if time.time() - os.path.getctime("data/hf_test/test_tree.root") > 86400.: # 24 hours in seconds
+            os.remove("data/hf_test/test_tree.root")
+            command = "root -l -b -q test/scripts/genTree.C+"
+            (ret,outRaw,errRaw) = script_runner(command)
+            assert ret.returncode == 0 # ROOT file with TTrees generated
 
     # Make input TTree if needed
-    if not os.path.isfile("test_tree.root"): 
-        command = "root -l -b -q ${HISTFITTER}/test/scripts/genTree.C+"
```

Comment:
I need a clear description of how you want these tests to run and where to save the output. Now each test will make a new folder in results/, config/, data/, etc, so it will look like this:
/config
  /BackupCacheExample
  /test_fullchain
  /MyUserAnalysis
Is this not the behaviour we want?

Reply:
My point is that you are removing the absolute paths ${HISTFITTER}/... and making them relative.  This means one can only run the tests from the top level directory. It's fine with me to update the tests to put things into various directories, but with relative paths, it now means that the files will pollute the standard directories.  I believe you were trying to avoid this.  With the full path to scripts using the env variable, the tests can be run in a temporary directory, which is easy to clean up.  So please keep the env variable so that the tests don't need to run in the top level directory.

Reply:
It's possible there is a mistake, but there are absolute paths using the env variable pointing to scripts that are in fixed locations, and then relative paths to the outputs.